lib/cc/cli/file_store.rb

```diff
@@ -0,0 +1,45 @@
+require "fileutils"
+require "yaml"
+
+module CC
+  module CLI
+    class FileStore
+      # This class is not supposed to be directly used. It should be sublcassed
+      # and a few constants need to be defined on the sublass to be usable.
+      #
+      # FILE_NAME is the name of the file this class wraps.
+
+      def initialize
+        load_data
+      end
+
+      def save
+        dir = File.dirname(self.class::FILE_NAME)
+        unless Dir.exist? dir
+          FileUtils.mkdir_p dir
```

Comment:
We shouldn't do this. If the files don't exist we should not attempt any version checking; and I wouldn't even have the code to do it, since it's a risk to write root-owned data.